### PR TITLE
Enabled torch.mm and torch.mv for bfloat16 on CPU

### PR DESCRIPTION
**Pull request by izdeby**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#24224 Enabled torch.mm and torch.mv for bfloat16 on CPU**

Differential Revision: [D16779996](https://our.internmc.facebook.com/intern/diff/D16779996)

----------

Enabled torch.mm and torch.mv for bfloat16 dtype. In order to do that, these methods have to be enabled for bfloat16.
- mul
- div
- addmm
- addmv
- addr

Also, overloaded arithmetic ops for BFloat16 including some bitwise ops such as '|', '&' and '^'

Tested via unit tests

### CODE REVIEW

File: test/test_torch.py [hunk 1]
```diff
@@ -377,7 +383,7 @@ def test_addmm(self):
                 for j in range(25):
                     for k in range(50):
                         res2[i, j] += m1[i, k] * m2[k, j]
-            self.assertEqual(res1, res2)
+            self.assertEqual(res1, res2, _prec)
```
Comment (VitalyFedyunin) [inline test/test_torch.py]: Usually underscore notation means unused variable, so you might want to rename it to `prec`.

**Review by VitalyFedyunin — COMMENTED** [review]
Will merge conflict with #23492 ( cc @pbelevich )

**Review by ezyang — CHANGES_REQUESTED** [review]
yeah, don't use `_prec` please

File: test/test_torch.py [hunk 2]
```diff
@@ -324,8 +326,9 @@ def test_addmv(self):
         types = {
             'torch.DoubleTensor': 1e-8,
             'torch.FloatTensor': 1e-4,
+            'torch.BFloat16Tensor': 0.4,
```
Comment (gchanan) [inline test/test_torch.py]: use the same format please.

File: test/test_torch.py [hunk 3]
```diff
@@ -348,14 +352,16 @@ def test_addmv(self):
...
                     res2[i] += m[i, j] * v[j]
-            self.assertEqual(res1, res2)
+
+            self.assertEqual(res1, res2, prec)
 
     def test_addmm(self):
         types = {
             'torch.DoubleTensor': 1e-8,
             'torch.FloatTensor': 1e-4,
+            'torch.BFloat16Tensor': 0.3,
```
Comment (gchanan) [inline test/test_torch.py]: use the  same format please.

**Review by gchanan — APPROVED** [review]

**Pull request closed without merging**